x/posts/keeper/handler.go

```diff
@@ -38,14 +38,15 @@ func NewHandler(keeper Keeper) sdk.Handler {
 
 // handleMsgCreatePost handles the creation of a new post
 func handleMsgCreatePost(ctx sdk.Context, keeper Keeper, msg types.MsgCreatePost) (*sdk.Result, error) {
+	postTime := ctx.BlockTime()
 	post := types.NewPost(
-		types.ComputeID(msg.CreationDate, msg.Creator, msg.Subspace),
+		types.ComputeID(postTime, msg.Creator, msg.Subspace),
```

Comment:
Before removing the `CreationDate` from the post, `ComputeID` made sure that the user could not create two different posts with the same creation date for the same subspace. 

Now that the creation time is the block time, however, this check should change. Suppose that a client app like Mooncake gathers all the posts to be sent to the chain every 30 seconds, and then uses a single transaction to broadcast them. If we would have to keep the `subspace + time + creator` unique key, each transaction would fail. Since all the posts are sent together, and the block is going to be only one, all of them would have the same creation date, the same subspace and the same creator. This will result in many very different posts having the same id. 

To solve this problem, I think that what we can do is change the `ComputeID` to be a method of the handler itself (since it's used only here) that accepts a `MsgCreatePost` and returns the hash of its JSON representation. This way we would make sure that for each block the message is unique. 

Initially I've thought about doing `sha256(time + creator + subspace + message)`. Then I thought that the message is optional, and also medias and the poll should be considered. For this reason, I think that `sha256(json(message))` is the best alternative to the current implementation to assure truly unique post ids.